source/server-discovery-and-monitoring/tests/unified/logging-replicaset.yml

```diff
@@ -0,0 +1,269 @@
+description: "replicaset-logging"
+
+schemaVersion: "1.14"
+
+runOnRequirements:
+  - topologies:
+      - replicaset
+    minServerVersion: "4.4" # awaitable hello
+
+createEntities:
+  - client:
+      id: &setupClient setupClient
+      useMultipleMongoses: false
+
+tests:
+  - description: "Topology lifecycle"
+    operations:
+      - name: createEntities
+        object: testRunner
+        arguments:
+          entities:
+            - client:
+                id: &client client
+                observeLogMessages:
+                  topology: debug
+                observeEvents:
+                  - topologyDescriptionChangedEvent
+      # ensure the topology has been fully discovered before closing the client.
+      - name: waitForEvent
+        object: testRunner
+        arguments:
+          client: *client
+          event:
+            topologyDescriptionChangedEvent: {}
+          count: 4
+      - name: close
+        object: *client
+    expectLogMessages:
+      - client: *client
+        ignoreLogMessages:
+          - data:
+              message: "Starting server monitoring"
+          - data:
+              message: "Server heartbeat started"
+          - data:
+              message: "Server heartbeat succeeded"
+          - data:
+              message: "Server heartbeat failed"
+        messages:
+          - level: debug
+            component: topology
+            data:
+              message: "Starting topology monitoring"
+              topologyId: { $$exists: true }
+          - level: debug
```

Comment:
similarly, could we add comments for these "changed" events indicating what the expected changes are? I know we can't assert on them, but for anyone debugging failures e.g. missing change events it would be helpful for them to know what we were expecting to receive here. (same applies in similar test for other topologies)